cub/test/test_device_batch_memcpy.cu

```diff
@@ -252,20 +252,42 @@ void RunTest(BufferOffsetT num_buffers,
     num_total_bytes += h_buffer_sizes[i];
   }
 
-  // Shuffle input buffer source-offsets
-  std::uint_fast32_t shuffle_seed = 320981U;
-  if (input_gen == TestDataGen::RANDOM)
-  {
-    h_buffer_src_offsets = GetShuffledBufferOffsets<BufferOffsetT, ByteOffsetT>(h_buffer_sizes,
-                                                                                shuffle_seed);
-    shuffle_seed += 42;
-  }
+  const bool is_integrated_gpu = IsIntegrated();
 
-  // Shuffle input buffer source-offsets
-  if (output_gen == TestDataGen::RANDOM)
   {
-    h_buffer_dst_offsets = GetShuffledBufferOffsets<BufferOffsetT, ByteOffsetT>(h_buffer_sizes,
-                                                                                shuffle_seed);
+    // Shuffle input buffer source-offsets
+    std::uint_fast32_t shuffle_seed = 320981U;
+
+    const std::size_t total_required_mem = 2 * sizeof(BufferOffsetT) * h_buffer_sizes.size()
+                                         + 2 * sizeof(ByteOffsetT) * h_buffer_sizes.size();
+
+    if (is_integrated_gpu && FreeGlobalMem() < total_required_mem)
```

Comment:
This is not consistent with the change below

We should either always compute `total_required_mem` in the same way or add the `Is_integrated_gpu` check